src/DataCollection.Shared/ViewModels/DestinationRelationshipViewModel.cs

```diff
@@ -97,44 +96,92 @@ public async Task LoadViewModel(RelatedFeatureQueryResult relatedFeatureQueryRes
+            }
+        }
 
         /// <summary>
-        /// Get all of the values in the related table to display when editing feature
+        /// Tracks whether a refresh of <see cref="OrderedAvailableValues"/> is in progress
         /// </summary>
-        private async Task GetAvailableValues()
+        public bool IsRefreshingValues
         {
-            var queryParams = new QueryParameters()
+            get => _isRefreshingValues;
+            set
             {
-                ReturnGeometry = false,
-                WhereClause = "1=1",
-            };
+                if (_isRefreshingValues != value)
+                {
+                    _isRefreshingValues = value;
+                    OnPropertyChanged();
+                    (RefreshValuesCommand as DelegateCommand)?.RaiseCanExecuteChanged();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Starts a refresh of the <see cref="OrderedAvailableValues"/> collection.
+        /// </summary>
+        public ICommand RefreshValuesCommand => _refreshValuesCommand ?? (_refreshValuesCommand = new DelegateCommand(parm =>
+		    {
+	            _ = RefreshAvailableValues(true).ContinueWith(res => LoadSelectedFeaturePopup());
+		    },
+            (param) =>
+            {
+                return !IsRefreshingValues;
+            })
+        );
+
+        /// <summary>
+        /// Loads <see cref="OrderedAvailableValues"/> with popups for all of the entries in the relevant table.
+        /// Values loaded from cache if <paramref name="invalidateCache"/> is false, otherwise cache is overwritten.
+        /// </summary>
+        private async Task RefreshAvailableValues(bool invalidateCache = false)
+        {
+            // Ensures that cache is only updated/checked by one Task at a time
+            await cacheMutex.WaitAsync();
 
             try
             {
+                if (!invalidateCache && CachedTableResults.ContainsKey(FeatureTable))
+                {
+                    OrderedAvailableValues = CachedTableResults[FeatureTable];
+                    return;
+                }
+                IsRefreshingValues = true;
+
+                var queryParams = new QueryParameters { ReturnGeometry = false, WhereClause = "1=1" };
+
                 // Query and load all related records
-                var featureQueryResult = await FeatureTable.QueryFeatures(queryParams);
-                var availableValues = new ObservableCollection<PopupManager>();
+                var featureQueryResult = (await FeatureTable.QueryFeatures(queryParams)).ToList();
+                var availableValues = new List<PopupManager>(featureQueryResult.Count);
 
                 foreach (ArcGISFeature result in featureQueryResult)
                 {
-                    if (result.LoadStatus != LoadStatus.Loaded)
-                    {
-                        await result.LoadAsync();
-                    }
+                    await result.LoadAsync();
```

Comment:
If you want to speed this up 20x in Online mode, consider using ServiceFeatureTable's batch-loading API -- something like this:
```cs
if (FeatureTable is ServiceFeatureTable sft)
{
    await sft.LoadOrRefreshFeaturesAsync(featureQueryResult);
}
else
{
    foreach (ArcGISFeature result in featureQueryResult)
    {
        await result.LoadAsync();
    }
}
foreach (ArcGISFeature result in featureQueryResult)
{
    availableValues.Add(new PopupManager(new Popup(result, result.FeatureTable.PopupDefinition)));
}
```